metaflow/plugins/datatools/s3/s3op.py

```diff
@@ -582,11 +582,23 @@ def list_prefix(self, prefix_url, delimiter=""):
                 # - the trailing slash is significant in S3
                 if "Contents" in page:
                     for key in page.get("Contents", []):
-                        url = url_base + key["Key"]
+                        key_path = key["Key"]
+                        # filter out sibling directories that share the same prefix
+                        if delimiter == "":  # recursive mode
+                            normalized_prefix = prefix_url.path
+                            if not normalized_prefix.endswith("/"):
+                                normalized_prefix += "/"
+                            # Only include keys that are actually under our directory
+                            if not (
```

Comment:
This filters out an object with key exactly equal to `ABC/` when listing prefix `ABC/`. In S3 key names can actually contain `/` so someone could create a file called `ABC/` (it's weird but valid)

Reply:
so is the recommended change just > to >= ?

Reply:
i think that would work